source/extensions/quic_listeners/quiche/quic_io_handle_wrapper.h

```diff
@@ -50,6 +50,14 @@ class QuicIoHandleWrapper : public Network::IoHandle {
     }
     return io_handle_.recvmsg(slices, num_slice, self_port, output);
   }
+  Api::IoCallUint64Result recvmmsg(absl::FixedArray<absl::FixedArray<Buffer::RawSlice>>& slices,
+                                   uint32_t self_port, RecvMsgOutput& output) override {
+    if (closed_) {
+      return Api::IoCallUint64Result(0, Api::IoErrorPtr(new Network::IoSocketError(EBADF),
+                                                        Network::IoSocketError::deleteIoError));
+    }
```

Comment:
When does this happen? Shouldn't this be an assert/programming error?

Reply:
No. This is the whole purpose of this IoHandle wrapper. Because QUIC connections share one UDP socket, when connection thinks it closed a socket, actually the socket is still open. If somehow a system call was made after the connection closed its socket, this call shouldn't be propagated to the actual IoHandle instance.

Reply:
But what is the calling code supposed to do when it gets a EBADF error? Wouldn't this be a calling code programming error and shouldn't we prevent this at that layer?

Reply:
This isn't an IO error though, this is a straight up programming error. Can we just ASSERT or RELEASE_ASSERT here?

Reply:
IMO any program using EBADF to signal disconnection is a pretty big code smell. I'm not a fan of doing this, but I won't argue the point further unless either @ggreenway or @alyssawilk have strong opinions on this.

Reply:
I think it's an API call - basically it depends if there's any read looping and if we want to check
if (connected()) {read())) everywhere like we do for network connection read-disable.  I'd be against RELEASE_ASSERT but I think stating in the API it's invalid to read from closed fd and ASSERTING to catch in test is fine.